### PR TITLE
vm/vmss: expose image plan related arguments

### LINKED ISSUE #4233: [VM] Certain VM scenarios require explicit specification of plan information
Initially reported by @huzferd in PR #4159.

Create Data Science VM using templates.
Create Image from DSVM.
Attempting to restore DSVM from Image yields the following error:
{
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMMarketplaceInvalidInput",
        "message": "Creating a virtual machine from Marketplace image requires Plan information in the request. OS disk name is 'test1_disk1'."
      }
    ]
  }
}

### PR DESCRIPTION
**Pull request by yugangw-msft**
Applicable on images with billing information. Fix #4233 
Note, CLI has existing logic to retrieve such information if available in the image, so essentially these new arguments will only be useful to create a vm/vmss from custom images captured by users themselves. 

```
Marketplace Image Plan Arguments
    --plan-name                        : Plan name.
    --plan-product                     : Plan product.
    --plan-promotion-code              : Plan promotion code.
    --plan-publisher                   : Plan publisher.
```

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by yugangw-msft** [comment]
@troydai, can you help review this PR? We have committed to external users that we will deliver through the upcoming release, so I prefer it can get in as early as possible.

**Comment by troydai** [comment]
Looking.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -235,6 +213,36 @@ def _parse_image_argument(namespace):
         raise CLIError(err.format(namespace.image, [x['urnAlias'] for x in images]))
 
 
+def _get_image_plan_info_if_exists(namespace):
+    try:
+        compute_client = _compute_client_factory()
+        if namespace.os_version.lower() == 'latest':
+            top_one = compute_client.virtual_machine_images.list(namespace.location,
```
Comment (troydai) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Please verify if the `top_one` instance has `plan` property. If so you can return the value directly here without fetching the image again on line 233.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Verified that the `list` still returns `VirtualMachineImageResource` which doesn't have the `plan` information, so we have to invoke a `get`

**Pull request merged**